src/App.tsx

```diff
@@ -15,48 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParams {
+  sortField: string;
+  reverseField: boolean;
+}
+
+function getPrepareGoods(
+  goods: string[],
+  { sortField, reverseField }: SortParams,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = getPrepareGoods(goodsFromServer,
+    { sortField, reverseField });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortType.alphabet })}
+          onClick={() => setSortField(SortType.alphabet)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SortType.length })}
+          onClick={() => setSortField(SortType.length)}
```

Comment:
make on function factory for both sort buttons and call in jsx with proper SortType
```
  const handleSort = (sortType: SortType) => ()  => setSortField(sortType)
```
```suggestion
          onClick={() => setSortField(SortType.length)}
```
```suggestion
          onClick={handleSort(SortType.length)}
```